### ISSUE TITLE
We should handle maximum length EF validation rules before saving

### ISSUE THREAD

**Issue by mariannk** [nopSolutions/nopCommerce#114]
Had an issue come to me yesterday where a category page couldn't be saved. Basically, the user was trying to update their meta keywords on the seo tab and went beyond the 400 character limit. Property: MetaKeywords Error: The field MetaKeywords must be a string or array type with a maximum length of '400' is the error thrown...however, the user gets the white "error screen" and they don't know what happened. 

Source: http://www.nopcommerce.com/boards/t/31729/validation-message-for-meta-keywords-exceeding-400-characters-not-firing.aspx 

We can use the following code to load validation rule (maximum length). Just refactor it to enumerate through all properties and saved entities (graph). Note that it can affect performance. 

```
var type = ObjectContext.GetObjectType(entity.GetType()); 
var typeName = ObjectContext.GetObjectType(entity.GetType()).AssemblyQualifiedName; 
int? maxLength = _context.GetColumnMaxLength(typeName, "PropertyName");
```

Also see work item #99

**Comment by fouimette** [comment]
Is there a reason why data annotations are not being used?

**Comment by skoshelev** [comment]
Closed #114
